### PR TITLE
[db-up] Fix bug for server name and enable ssl enforecement

### PR DESCRIPTION
**Pull request by Juliehzl**
---
Fix https://github.com/Azure/azure-cli/issues/11633 , https://github.com/Azure/azure-cli/issues/11635, https://github.com/Azure/azure-cli/issues/12340

The PR will change the behavior for some commands:
1. ssl_enforcement will be enabled by default
2. When resource group name in namespace is different from what in cache, reset it and create new server name
3. Bump psycopg2 in MacOS, refering https://github.com/Azure/azure-cli-extensions/pull/1675/files

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S167

File: src/db-up/azext_db_up/_validators.py [hunk 1]
```diff
@@ -30,43 +30,59 @@ def db_down_namespace_processor(db_type):
...
 def _process_db_up_namespace(cmd, namespace, db_type=None):
-    # populate from cache if existing
-    _set_value(db_type, namespace, 'location', 'location', cache=False)
-    _set_value(db_type, namespace, 'resource_group_name', 'group', cache=False)
+    # populate from cache if existing when no resource group name provided
+    resource_client = resource_client_factory(cmd.cli_ctx)
+    if namespace.resource_group_name is None:
+        _set_value(db_type, namespace, 'resource_group_name', 'group', cache=False)
+        try:
+            resource_client.resource_groups.get(namespace.resource_group_name)
```
Comment (zhoxing-ms) [inline src/db-up/azext_db_up/_validators.py]: Why is there no operation to assign the returned result to a variable?

File: src/db-up/azext_db_up/_validators.py [hunk 2]
```diff
@@ -30,43 +30,59 @@ def db_down_namespace_processor(db_type):
...
+        _set_value(db_type, namespace, 'location', 'location', cache=False)
+    # generate smart defaults when namespace.location is None
     if _get_value(db_type, namespace, 'location', 'location') is None:
         try:
             get_default_location_from_resource_group(cmd, namespace)
         except (CLIError, ValidationError):
-            namespace.location = 'westus2'
-    _set_value(db_type, namespace, 'location', 'location', namespace.location)
+            namespace.location = 'eastus'
+    _set_value(db_type, namespace, 'location', 'location', default=namespace.location)
```
Comment (zhoxing-ms) [inline src/db-up/azext_db_up/_validators.py]: Why is it changed from `westus2` to `eastus`

File: src/db-up/azext_db_up/util.py [hunk 3]
```diff
@@ -30,8 +30,12 @@ def create_random_resource_name(prefix='azure', length=15):
...
     config_section = CONFIG_MAP[db_type]
     if fallback == '_fallback_none':
-        return DB_CONFIG.get(config_section, option)
-    return DB_CONFIG.get(config_section, option, fallback=fallback)
+        result = DB_CONFIG.get(config_section, option)
+    else:
+        result = DB_CONFIG.get(config_section, option, fallback=fallback)
+    if result:
+        return result
+    return None
```
Comment (zhoxing-ms) [inline src/db-up/azext_db_up/util.py]: If the `result` is `0` or `False`, the result of function returns None, is that as expected?

File: src/db-up/azext_db_up/_validators.py [hunk 4]
```diff
@@ -30,43 +30,59 @@ def db_down_namespace_processor(db_type):
...
+    if namespace.resource_group_name is None:  # No resource group provided and in cache
         namespace.resource_group_name = create_random_resource_name('group')
     else:
         try:
             resource_client.resource_groups.get(namespace.resource_group_name)
             create_resource_group = False
-        except CloudError:
-            pass
+        except CloudError as ex:  # throw exception when resource group name is invalid
+            raise ex
```
Comment (qianwens) [inline src/db-up/azext_db_up/_validators.py]: just remove the try except

Comment (zhoxing-ms) [inline src/db-up/azext_db_up/_validators.py]: Below is the logic of create resource group. The original logic was that if the resource group was not found in the Service, the resource group was created on the Service and the rest of the process was followed.
Is it in line with the expectation to directly exit the following process?

**Comment by Juliehzl** [comment]
It is hard to record such scenario with playback test, so make them live_only. And all tests passed locally:
image

File: src/db-up/azext_db_up/util.py (hunk 3 above)
Comment (Juliehzl) [reply src/db-up/azext_db_up/util.py:38]: yes

File: src/db-up/azext_db_up/_validators.py (hunk 2 above)
Comment (Juliehzl) [reply src/db-up/azext_db_up/_validators.py:53]: westus2 will be not available in some cases.

File: src/db-up/azext_db_up/_validators.py (hunk 1 above)
Comment (Juliehzl) [reply src/db-up/azext_db_up/_validators.py:40]: yes. Here is to check if the resource group name in cache is still available.

**Pull request merged**